src/components/MainViewer/MainViewer.js

```diff
@@ -115,46 +117,43 @@ function MainViewer({
       );
       if (secRef.current !== tempSec) {
         secRef.current = tempSec;
-        throttledDraw([getIdChArr()]);
+        throttledSetImgState(getIdChArr(), width, height);
       }
     }
   };
 
-  const draw = useCallback(
-    async (idChArr) => {
-      if (idChArr.reduce((sum, x) => sum + x.length, 0) === 0) return;
-      const [images, promise] = getSpecWavImages(
-        idChArr[0],
+  const throttledSetImgState = useCallback(
+    throttle(1000 / 240, (idChArr, width, height) => {
+      if (idChArr.length === 0) return;
+      setImgState(
+        idChArr,
         secRef.current,
         width,
         {...drawOptionRef.current, height: height},
         {min_amp: -1, max_amp: 1},
+        0.3,
       );
+    }),
+    [],
+  );
 
-      for (const [idChStr, bufs] of Object.entries(images)) {
-        const ref = children.current[idChStr];
-        // let promises = [];
-        if (ref) {
-          // promises.push(
-          ref.draw(bufs);
-          // );
-        }
-        // Promise.all(promises);
-      }
-
-      // cached image
-      if (promise !== null) {
-        const arr = await promise;
-        debouncedDraw(arr);
+  const draw = (_) => {
+    const images = getImages();
+    let promises = [];
+    for (const [idChStr, buf] of Object.entries(images)) {
+      const ref = children.current[idChStr];
+      if (ref) {
+        promises.push(ref.draw(buf));
       }
-    },
-    [height, width],
-  );
+    }
+    Promise.all(promises);
+    requestRef.current = requestAnimationFrame(draw);
+  };
 
-  const throttledDraw = useCallback(throttle(1000 / 60, draw), [draw]);
-  const debouncedDraw = useCallback(debounce(1000 / 30, draw), [draw]);
-  // const throttledDraw = draw;
-  // const debouncedDraw = draw;
+  useEffect(() => {
+    requestRef.current = requestAnimationFrame(draw);
+    return () => cancelAnimationFrame(requestRef.current);
+  }, []);
```

Comment:
Hooks should go to the bottom of the code

Reply:
suggestion accepted.